src/Http/Cookie/Cookie.php

```diff
@@ -0,0 +1,506 @@
+    }
+
+    /**
+     * Check if the cookie is secure
+     *
+     * @return bool
+     */
+    public function isSecure()
+    {
+        return $this->secure;
+    }
+
+    /**
+     * Set HTTP Only
+     *
+     * @param bool $httpOnly HTTP Only
+     * @return $this
+     */
+    public function setHttpOnly($httpOnly)
+    {
+        if (!is_bool($httpOnly)) {
+            throw new InvalidArgumentException(sprintf(
+                'The provided arg must be of type `bool` but `%s` given',
+                gettype($httpOnly)
+            ));
+        }
+
+        $this->httpOnly = $httpOnly;
+
+        return $this;
+    }
+
+    /**
+     * Check if the cookie is HTTP only
+     *
+     * @return bool
+     */
+    public function isHttpOnly()
+    {
+        return $this->isHttpOnly();
+    }
+
+    /**
+     * Sets the expiration date
+     *
+     * @param \DateTimeInterface $dateTime Date time object
+     * @return $this
+     */
+    public function expiresAt(DateTimeInterface $dateTime)
+    {
+        $this->expiresAt = (int)$dateTime->format('U');
+
+        return $this;
+    }
+
+    /**
+     * Checks if a value exists in the cookie data
+     *
+     * @param string $path Path to check
+     * @return bool
+     */
+    public function check($path)
+    {
+        return Hash::check($this->value, $path);
+    }
+
+    /**
+     * Writes data to the cookie
+     *
+     * @param string $path Path to write to
+     * @param mixed $value Value to write
+     * @return $this
+     */
+    public function write($path, $value)
+    {
+        if (!$this->isExpanded) {
+            throw new RuntimeException('The Cookie data has not been expanded');
+        }
+
+        Hash::insert($this->value, $path, $value);
+
+        return $this;
+    }
+
+    /**
+     * Read data from the cookie
+     *
+     * @param string $path Path to read the data from
+     * @return mixed
+     */
+    public function read($path = null)
+    {
+        if (!$this->isExpanded) {
+            throw new RuntimeException('The Cookie data has not been expanded');
+        }
+
+        if ($path === null) {
+            return $this->value;
+        }
+
+        return Hash::get($this->value, $path);
+    }
+
+    /**
+     * Sets the cookies date to a far future so it will virtually never expire
+     *
+     * @return $this
+     */
+    public function willNeverExpire()
+    {
+        $this->expiresAt = Chronos::now()->setDate(2038, 1, 1)->format('U');
+
+        return $this;
+    }
+
+    /**
+     * Deletes the cookie from the browser
+     *
+     * This is done by setting the expiration time to "now"
+     *
+     * @return $this
+     */
+    public function willBeDeleted()
+    {
+        $this->expiresAt = Chronos::now()->format('U');
+
+        return $this;
+    }
+
+    /**
+     * Encrypts the cookie value
+     *
+     * @param string $key Encryption key
+     * @return $this
+     */
+    public function encrypt($key)
```

Comment:
Should `$key` be nullable? Now `setEncryptionKey()` seems not necessary as `encryptionKey` gets overwritten anyway in `encrypt()`/`decrypt()`.